common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -52,6 +52,9 @@ def student_view(self, context):
         fragment.add_content(self.system.render_template('vert_module.html', {
             'items': contents,
             'xblock_context': context,
+            'bookmarked': child_context['bookmarked'],
+            'item_id': unicode(self.location),
+            'username': child_context['username']
```

Comment:
I think it would be simpler to just do `bookmark_id = "{},{}".format(child_context['username'],unicode(self.location))` and use that instead of constructing the bookmark_id in javascript from these two.